xcm/xcm-simulator/example/src/lib.rs

```diff
@@ -212,4 +212,39 @@ mod tests {
 			);
 		});
 	}
+
+	#[test]
+	fn recursion_limit_works() {
+		use parachain::{Event, System};
+		MockNet::reset();
+
+		ParaA::execute_with(|| {
+			let msg = WithdrawAsset { assets: vec![All], effects: vec![] };
+			let msg = RelayedFrom { who: Null, message: Box::new(msg) };
```

Comment:
should be a loop which uses the recursion constant.

Shows that `MAX_RECURSION_LIMIT - 1` works, and `MAX_RECURSION_LIMIT` doesnt